### DISCUSSION TITLE
Bug: Rebooting the Docker Container Results in 31,519 hours of Visit Duration

### DISCUSSION THREAD

**Discussion by Makosai** [plausible/analytics#4099] [Self-Hosted Support] [answered]
![image](https://github.com/plausible/analytics/assets/7884968/260a50de-54cc-4c5e-ba4c-a470a2c80c80)

What I did:
1. Lowered the clickhouse-server version from 24.3.3.102-alpine to 23.3.7.5-alpine
2. Raised the analytics version from 2.1.0-rc.0 to 2.1.0-rc.1

```sh
$ docker compose stop plausible plausible_events_db
$ docker compose rm plausible plausible_events_db
$ docker compose up -d
```

Is it possible to remedy or delete this entry? Whether it's manually or automatically?

**Comment by ruslandoga** [comment, marked as answer]
👋 @Makosai 

Sessions are kept in memory and can be indefinitely extended if events for it arrive within 30 minutes of each other. A restart causes all sessions to be "flushed". You probably had something like a health-check processes creating a very long session. You can identify it in ClickHouse and drop it manually.

```console
$ docker compose exec plausible_events_db clickhouse client --database plausible_events_db
```
```sql
-- identify the session you want to delete
:) select session_id, formatReadableTimeDelta(max(timestamp) - min(start)) as visit_duration from sessions_v2 group by session_id order by max(timestamp) - min(start) desc limit 10;
-- delete the session
:) delete from sessions_v2 where session_id=xxx;
-- delete the events
:) delete from events_v2 where session_id=xxx;
```

**Comment by Makosai** [comment]
Hi @ruslandoga! Thanks a ton for the reply.

Here's my list of sessions -- I also increased the size of the list:
![image](https://github.com/plausible/analytics/assets/7884968/5c7bb8b9-321b-4b10-af65-8326e005396f)

I did remove one that was 56 minutes long earlier. This is the result. But that change in the number could also be because it started to average out the rest of the day:
![image](https://github.com/plausible/analytics/assets/7884968/bd35c4a0-daaf-46ed-91cd-7de852b39529)

How much should I delete?

**Comment by Makosai** [comment]
![image](https://github.com/plausible/analytics/assets/7884968/b8a4342a-5673-4a14-b519-dddd1cbb19a7)

Updating the database back to 24.3.3.102-alpine fixed it apparently. So this is resolved. If you want to investigate any of it, I'm willing to prod around to see why.
